vivo-project/VIVO
# Issue/vivo 1908

**[VIVO-1908](https://jira.lyrasis.org/browse/VIVO-1908)**:

* Also addresses subtask https://jira.lyrasis.org/browse/VIVO-1906

# What does this pull request do?
Uses the existing language-filtering mechanism for generating dropdown select lists with language-appropriate labels on editing forms (such as "organizer of event") where the URIs are in the select lists are harcoded in the form.  Also adds a fallback mechanism to the manage webpages for individual form where language filtering is injected into a SPARQL query.

# What's new?
See GeneratorUtil.java and ManageWebapagesForIndividualGenerator.java for the most substantive changes.  
GeneratorUtil now has a method buildResourceAndLabelFieldOptions() that is very similar to the original method for building a list of constant field options.  This new method uses a WebappDaoFactory, if available, to retrieve in-memory class labels, and an RDFService to retrieve labels for all other URIs.  The individual editing forms supply the existing WebappDaoFactory and RDFService from the request, which already perform language filtering.  The editing forms no longer build DESCRIBE queries where language filtering is injected.

ManageWebpagesForIndividual.java adds fallback languages to the SPARQL query that retrieves webpage links as well as the labels for the link types.  Because this query involves GROUP BY, language filtering here cannot be handled by the RDFService-level filtering.

# How should this be tested?
* Create/navigate to a person.  From the Service tab, select "organizer of event".  Change the language selector, and note that event type labels in the dropdown continue to match the selected language.  For more in-depth testing, add a new URI to the list in AddOrganizerRoleToPerson.java, create that class in VIVO and supply a label in only one language, and note that the class continues to appear in the dropdown even if a label is not available that matches your current language selection.
* Go to the class hierarchy, select all classes, find "F1000 Link", and add a new subclass of that class.  Supply a label in only one language.  Navigate to a person.  Open the manage webpages page (Rolodex) and enter a new webpage of the new type you just created.  From the manage webpages page, change the language selector and note that the label for the new type continues to appear even if a label does not exist that matches the current language selection.
